src/test/java/com/fittracker/fittracker/exception/ErrorResponseMapperTest.java

```diff
@@ -0,0 +1,200 @@
+package com.fittracker.fittracker.exception;
+
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.MissingServletRequestParameterException;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ErrorResponseMapperTest {
+
+    @InjectMocks
+    private ErrorResponseMapper errorResponseMapper;
+
+    private final LocalDate TEST_DATE = LocalDate.of(2023,11,10);
+
+
+    BindingResult bindingResult;
+    FieldError fieldError;
+    MethodParameter methodParameter;
+    HttpMessageNotReadableException httpMessageNotReadableException;
+
+    @BeforeEach
+    void setUp() {
+        bindingResult = mock(BindingResult.class);
+        fieldError = mock(FieldError.class);
+        methodParameter = mock(MethodParameter.class);
+        httpMessageNotReadableException = mock(HttpMessageNotReadableException.class);
+    }
+
+    @Test
+    void givenWeightNotFoundException_shouldReturnErrorResponse() {
+
+        try {
+            throw new WeightNotFoundException(TEST_DATE);
+        } catch (WeightNotFoundException e) {
+            var expectedErrorResponse = new ErrorResponse(null, "Weight not found for date: 2023-11-10");
+            var errorResponse = errorResponseMapper.map(e);
+            assertThat(errorResponse).usingRecursiveComparison().isEqualTo(expectedErrorResponse);
```

Comment:
`usingRecursiveComparison` is not needed here, because records by default override `equals` method, comparing objects field by field